wordpress/wp-content/plugins/memberful-wp/src/contrib/woocommerce.php

```diff
@@ -18,15 +18,27 @@ public static function instance() {
   function __construct() {
     add_action( 'woocommerce_single_product_summary', array( $this, 'hide_add_to_cart_button' ), 25);
     add_filter( 'woocommerce_add_to_cart_validation', array( $this, 'block_cart_add' ), 30, 3 );
-    add_filter( 'woocommerce_is_purchasable', array( $this, 'not_purchasable'), 20, 2 );
+    add_filter( 'woocommerce_is_purchasable', array( $this, 'is_purchasable'), 20, 2 );
+    add_filter( 'the_content', array( $this, 'remove_erroneous_protection' ), -20 );
+  }
+
+  /**
+   * Removes filtering on the content from other functionalities
+   */
+  function remove_erroneous_protection( $content ) {
+    global $post;
+    if ($post->post_type == "product") {
+      remove_filter( 'the_content', 'memberful_wp_protect_content', -10 );
+    }
+    return $content;
   }
 
   /**
    * Makes it so that the add to cart button on archive page will show Read More and link to product on
    * Archive (loop) pages
    */
-  function not_purchasable( $purchasable, $product ) {
-    return !memberful_can_user_access_post( get_current_user_id(), $product->get_id() ) ? false : $purchasable;
+  function is_purchasable( $purchasable, $product ) {
+    return !current_user_can('administrator') && !memberful_can_user_access_post( get_current_user_id(), $product->get_id() ) ? false : $purchasable;
```

Comment:
`current_user_can("administrator")` doesn't work well with multisite mode according to [this stack overflow question](https://wordpress.stackexchange.com/questions/199694/current-user-canadministrator-does-not-return-true-in-multisite-if-user-is-a). Could you please replace it with `memberful_is_admin()`? We already use it in `src/syncing.php`.

Reply:
OK I changed it to use that function - https://github.com/memberful/memberful-wp/pull/222/commits/90475736b7181b9de8e514bd00a0995fb96a470a